packages/typedoc-plugin-appium/lib/output/theme/appium.ts

```diff
@@ -0,0 +1,61 @@
+import {ContainerReflection, PageEvent, Renderer} from 'typedoc';
+import {MarkdownTheme} from 'typedoc-plugin-markdown';
+import {AppiumPluginLogger} from '../../logger';
+import {AppiumPluginReflectionKind} from '../../model';
+import {compileTemplate, registerHelpers, Template} from './utils';
+
+/**
+ * Name of the theme; used at definition time
+ */
+export const THEME_NAME = 'appium';
+
+/**
+ * Factory for `AppiumTheme` class; needs custom logger otherwise inaccessible
+ * @param log - Custom logger
+ * @returns `AppiumTheme` class
+ */
+export function getTheme(log: AppiumPluginLogger): new (renderer: Renderer) => MarkdownTheme {
+  return class AppiumTheme extends MarkdownTheme {
```

Comment:
I know this is a correct syntax, but `return class` is still confusing to me

Reply:
I suppose I should explain why I did this.

We have this `AppiumPluginLogger` instance, right.  and the instance is created _from an instance of the default `Logger`_ in `typedoc`--it kind of wraps the instance.  There's no easy & good way--short of a global singleton instance of `AppiumPluginLogger`--to provide it to an `AppiumPluginTheme`.  TypeDoc does the instantiation of the theme class, so I'm not able to just pass arbitrary arguments to its constructor.

There's a "component" architecture in TypeDoc that I could maybe leverage, but it is apparently being phased-out (per the developer docs of TypeDoc).  It looked overcomplicated, so I didn't want to touch it.

Reply:
I may also have been able to do this via the event system (`this.emit('here-is-a-logger', log)`), but that seems like abuse of the event system.

Reply:
maybe we could just define class separately, not in the same line with return, to make that easier to read